lib/client.js

```diff
@@ -302,6 +302,64 @@ class Client {
     return Promise.resolve(this.__dataApiClient)
   }
 
+  /**
+   * Returns a decoded Avro Promise given the proper schema name and data
+   * @param {String} schemaName
+   * @param {Object|String} data
+   * @returns {Promise}
+   */
+  decodeData (schemaName, data) {
+    if (!schemaName || typeof schemaName !== 'string' || schemaName.trim() === '') {
+      log.debug('decodeData() schemaName parameter not defined');
+      return null;
```

Comment:
should this throw null or throw an exception? Not sure what is the node convention of handling this. Throwing an exception is like a programming practice, right?